### ISSUE TITLE
aws_elasticache_parameter_group crashes terraform

### ISSUE THREAD

**Issue by jenarros** [hashicorp/terraform#7177]
### Terraform Version
Terraform v0.6.15

### Affected Resource(s)
- aws_elasticache_parameter_group
i.e.:
```
resource "aws_elasticache_parameter_group" "events_redis_params" {
  name = "events-redis-params"
  family = "redis2.8"
  description = ""

  parameter {
    name = "maxmemory-policy"
    value = "allkeys-lru"
  }

  parameter {
    name = "maxmemory-samples"
    value = "20"
  }
}
```

### Debug Output
https://gist.github.com/jenarros/6a687adaa8075da4a8205b27ea12ac66

### Panic Output
https://gist.github.com/jenarros/6a687adaa8075da4a8205b27ea12ac66

### Expected Behavior
Terraform shows whether there are any changes between the state file and the aws resource above

### Actual Behavior
Terraform crashes

### Steps to Reproduce
1.- Create Redis elasticache with parameter group as above (see example in Affected Resources)
2.- Modify the maxmemory-samples parameter using AWS UI to a different value
3.- Run terraform plan

**Comment by stack72** [comment]
Hi @jenarros! We believe this bug has been fixed in the latest version of Terraform. Could you try your reproduction with 0.6.16 and verit

**Comment by jenarros** [comment]
Hi, I have tried with 0.6.16 and I can see the same problem. It seems to be related to issue https://github.com/hashicorp/terraform/pull/6687 but it does not seem like the fix was merged into 0.6.16, do you know when version 0.7 will be released? Kind Regards

**Comment by stack72** [comment]
Hi @jenarros 

just tested this on 0.7 rc2 and it works as expected.

Terraform apply of your code:

```
7177 % terraform apply
aws_elasticache_parameter_group.events_redis_params: Creating...
  family:                     "" => "redis2.8"
  name:                       "" => "events-redis-params"
  parameter.#:                "" => "2"
  parameter.2842875953.name:  "" => "maxmemory-samples"
  parameter.2842875953.value: "" => "20"
  parameter.395915293.name:   "" => "maxmemory-policy"
  parameter.395915293.value:  "" => "allkeys-lru"
aws_elasticache_parameter_group.events_redis_params: Creation complete

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```

When i change the UI to have `maxmemory-samples = 40`, I terraform refresh and see the following

```
7177 % terraform refresh
aws_elasticache_parameter_group.events_redis_params: Refreshing state... (ID: events-redis-params)
```

I am not 100% sure when 0.7 will be released. I think that is a question for the core team (@phinze / @jen20 / @catsby) - hopefully they will be able to help you here :)

Paul

**Comment by jenarros** [comment]
Closing as it seems that it has been fixed in 0.7
